src/app/core/services/models/ntp-event-observer.ts

```diff
@@ -0,0 +1,38 @@
+import { EventConsumer, EventObserver, EventSubscription, TypedEvent } from '../../../event-system/abstractions/event-observer.service'
+import { Logger } from '../../abstractions/logger.service'
+import { NtpEventParser } from '../../abstractions/ntp-event-parser'
+import { NtpEvent } from '../../models/ntp-event'
+import { NtpEventType } from '../../enums/ntp-event-type'
+import { Injectable } from '@angular/core'
+
+@Injectable()
+export class NtpEventObserver {
+  private readonly logger: Logger
+
+  constructor(
+    private readonly eventObserver: EventObserver,
+    private readonly ntpEventParser: NtpEventParser,
+    logger: Logger
+  ) {
+    this.logger = logger.tag('NtpEventObserver')
```

Comment:
```suggestion
    this.logger = logger.tag(this.constructor.name)
```

Just a suggestion :) There are multiple places in the project where we could do this.

Reply:
In the frontend we don't use `this.constructor.name` since it is obscured when bundled due to variable collision prevention and size reduction :)